subprojects/language-java/src/integTest/groovy/org/gradle/java/compile/incremental/SourceIncrementalJavaCompilationIntegrationTest.groovy

```diff
@@ -569,6 +569,25 @@ sourceSets {
         output.contains("Full recompilation is required because the source roots could not be inferred.")
     }
 
+    def "missing files are ignored of source roots"() {
+        buildFile << "compileJava.source([file('extra-java'), file('other'), fileTree('missing-tree'), file('missing-file')])"
+
+        java("class A extends B {}")
+        file("extra-java/B.java") << "class B {}"
+        file("extra-java/C.java") << "class C {}"
+
+        outputs.snapshot { run "compileJava" }
+
+        when:
+        file("extra-java/B.java").text = "class B { String change; }"
+        executer.withArgument "--info"
+        run "compileJava"
+
+        then:
+        outputs.recompiledClasses("A", "B")
+        !output.contains("Full recompilation is required because the source roots could not be inferred.")
```

Comment:
It's not an isolated problem, but I feel such tests are way too brittle. Can we at least get this string from a constant in the production code maybe?

Reply:
I can remove this line, given that a full recompilation would always meant the `C` is compiled as well.